templates/CRM/Contact/Form/Edit/Address.tpl

```diff
@@ -22,7 +22,7 @@
  <div id="Address_Block_{$blockId}" {if $className eq 'CRM_Contact_Form_Contact'} class="boxBlock crm-edit-address-block crm-address_{$blockId}"{/if}>
   {if $blockId gt 1}<fieldset><legend>{ts}Supplemental Address{/ts}</legend>{/if}
   <table class="form-layout-compressed crm-edit-address-form">
-     {if $masterAddress.$blockId gt 0 }
+     {if isset($masterAddress) && $masterAddress.$blockId gt 0 }
```

Comment:
I'll be you anything the IDs don't go into negative numbers and the `gt` was just another way of saying
```suggestion
     {if !empty($masterAddress.$blockId) }
```
But your version is a more literal interpretation of what was already there so probably safer.